ui/shared/AddressInput.qml

```diff
@@ -11,47 +11,63 @@ Item {
     //% "ENS Username not found"
     property string ensAsyncValidationError: qsTrId("ens-username-not-found")
     property alias label: inpAddress.label
+    property alias text: inpAddress.text
     property string selectedAddress
     property var isValid: false
     property bool isPending: false
+    readonly property string uuid: Utils.uuid()
+    property alias readOnly: inpAddress.readOnly
+    property bool isResolvedAddress: false
 
     height: inpAddress.height
 
+    onSelectedAddressChanged: validate()
+    onTextChanged: resolveEns()
+
     function resetInternal() {
         selectedAddress = ""
         inpAddress.resetInternal()
         metrics.text = ""
         isValid = false
+        isPending = false
+        isResolvedAddress = false
+    }
+
+    function resolveEns() {
+        if (Utils.isValidEns(text)) {
+            root.validateAsync(text)
+        }
     }
 
-    function validate(inputValue) {
-        if (!inputValue) inputValue = selectedAddress
-        let isValid =
-            (inputValue && inputValue.startsWith("0x") && Utils.isValidAddress(inputValue) || Utils.isValidEns(inputValue))
-        inpAddress.validationError = isValid ? "" : validationError
+    function validate() {
+        let isValidEns = Utils.isValidEns(text)
+        let isValidAddress = Utils.isValidAddress(selectedAddress)
+        let isValid = (isValidEns && !isResolvedAddress) || isPending || isValidAddress
+        inpAddress.validationError = ""
+        if (!isValid){ 
+            inpAddress.validationError = isResolvedAddress ? ensAsyncValidationError : validationError
+        }
         root.isValid = isValid
         return isValid
     }
 
-    property var validateAsync: Backpressure.debounce(inpAddress, 300, function (inputValue) {
+    property var validateAsync: Backpressure.debounce(inpAddress, 600, function (inputValue) {
```

Comment:
Could be wrong, but debouncing 600ms is a little long. Might be enough to get users wonder why nothing's happening.